engine/src/test/java/org/hibernate/validator/test/spi/nodenameprovider/jackson/JacksonAnnotationPropertyNodeNameProvider.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.test.spi.nodenameprovider.jackson;
+
+import org.hibernate.validator.spi.nodenameprovider.Property;
+import org.hibernate.validator.spi.nodenameprovider.PropertyNodeNameProvider;
+
+import com.fasterxml.jackson.databind.BeanDescription;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.introspect.BeanPropertyDefinition;
+
+public class JacksonAnnotationPropertyNodeNameProvider implements PropertyNodeNameProvider {
```

Comment:
@marko-bekhta this is an example of `PropertyNodeNameProvider` for the [jackson](https://github.com/FasterXML/jackson) use-case (which will probably be the most used one). Here we can see that the implementor has the freedom of using any API that can assist him in implementing the correct behaviour. In this example, jackson provides all the necessary API-s that are needed for the implementation. If `internal.properties.*` becomes open one day, the implementor can use it.

The point is that this feature, in our opinion, is not related, or blocked by `internal.properties.*` being opened or closed.

[this](/engine/src/test/java/org/hibernate/validator/test/spi/nodenameprovider/jackson/JacksonAnnotationPropertyNodeNameProviderTest.java) is a test for `JacksonAnnotationPropertyNodeNameProvider `

Reply:
So, indeed you can work around not having the properties as you did it.

But I think it will be very slow and probably not acceptable in the general case. Take a look at what you have to do in the `getName()` method. It does not look good performance-wise.

While I don't think we're ready to push `internal.properties.*` in SPIs, I think it might be possible to have a subset of it covering the most common use cases (e.g. property name and annotations). Could you maybe take a stab at it?

Reply:
@gsmet: i agree that it might be a bit of a hassle for some the cases to implement something simple and that we should provide more property related information. but this is why we proposed to have the `Property` interface, to be flexible with new features that improve the experience when implementing `PropertyNodeNameProvider`.

I just wanted to indicate that e.g. `Jackson` is more complicated than it seems. Influencing how a property is resolved cannot only be done through an annotation but also through a globally defined naming strategy. Plus `Jackson` is doing a lot of caching, the call that is made (as an example) in the `JacksonAnnotationPropertyNodeNameProvider.getName` is actually quite fast.

So i would suggest to refactor `PropertyNodeNameProvider.getName` to:
```
String getName(Property property);
```

and the `Property` to:

```
public interface Property {
	String getName();
	Annotation[] getDeclaredAnnotations();
	Class<?> getDeclaringClass();
	Object getObject();
}
```

WDYT?